### ISSUE TITLE
Move the deprecated random_state alias from AnvilSection to the classes that own random_seed

### ISSUE THREAD

**Issue by smcolby** [OpenADMET/openadmet_models#596]
Components expose `random_seed`, and `random_state` is accepted as a deprecated alias for it. That alias is honored only when constructing from a spec. `AnvilSection._map_deprecated_random_state` (`anvil/specification.py:435`) is a `model_validator(mode="before")` that rewrites the key inside a section's `params` dict; it is the only implementation. No class declares `random_state` as a field or handles the alias itself.

Constructing the same component directly in Python does not map the alias, and the supplied seed is lost.

## Observed behavior

Passing `random_seed` works everywhere. All ten scikit-learn-style models hand the value to their estimator correctly, on both paths.

Passing `random_state` diverges by path:

| construction | result |
|---|---|
| `SplitSpec(type="ScaffoldSplitter", params={"random_state": 7}).to_class()` | `random_seed=7`, `DeprecationWarning` raised |
| `ScaffoldSplitter(random_state=7)` | `random_seed=42` (the default), no warning |

For models the direct path is worse than a dropped value, because the estimator is actively unseeded. `seed_to_sklearn_kwargs` (`architecture/model_base.py:17`) keys off the *presence* of `random_seed`, not its value, and `model_dump()` always contains it:

```python
seed_to_sklearn_kwargs({"random_seed": None, "random_state": 9})  # -> {"random_state": None}
```

So a caller-supplied `random_state` is overwritten with `None` at build time. This holds regardless of the model's `extra` setting: `extra="allow"` keeps the key through validation, and it is then clobbered anyway.

Constructing each model with `random_state=9` and inspecting the built estimator:

| model | estimator seed |
|---|---|
| `RFRegressorModel`, `RFClassifierModel` | `None` |
| `SVMClassifierModel`, `DummyClassifierModel` | `None` |
| `LGBMRegressorModel`, `LGBMClassifierModel` | `None` |
| `XGBRegressorModel`, `XGBClassifierModel` | `None` |
| `CatBoostRegressorModel`, `CatBoostClassifierModel` | `9` |

The CatBoost models are unaffected because CatBoost's native argument is already `random_seed`, so they never route through the helper.

The result is a run that reads as seeded and is not, with no error and no warning.

## Further problems

**The `DeepLearningFeaturizer` docstring is inaccurate.** `features/feature_base.py:99-101` states that "the legacy `random_state` name is accepted as a deprecated alias." That is true from a spec and false from direct construction.

**The mapping fires for classes that have no `random_seed`.** `TrainerSpec(type="SKLearnBasicTrainer", params={"random_state": 7})` rewrites the key and warns, directing the caller to a field `SKLearnBasicTrainer` does not define; the value is discarded at construction.

**The generic section wrapper knows a domain-specific field name.** `AnvilSection` is the shared `{type, params}` container for splits, models, featurizers, trainers, evaluators, and transforms. Rewriting one particular key inside an opaque `params` dict is not its concern; the classes that declare `random_seed` should own their own deprecated alias.

## Scope

24 registered classes declare `random_seed`, with no shared seed-owning base:

- **split** (5): `ClusterSplitter`, `MaxDissimilaritySplitter`, `PerimeterSplitter`, `ScaffoldSplitter`, `ShuffleSplitter`
- **model** (14): `CatBoostClassifierModel`, `CatBoostRegressorModel`, `DummyClassifierModel`, `LGBMClassifierModel`, `LGBMRegressorModel`, `RFClassifierModel`, `RFRegressorModel`, `SVMClassifierModel`, `TabPFNClassifierModel`, `TabPFNPostHocClassifierModel`, `TabPFNPostHocRegressorModel`, `TabPFNRegressorModel`, `XGBClassifierModel`, `XGBRegressorModel`
- **transform** (2): `ImputeTransform`, `PCATransform`
- **eval** (2): `SKLearnRepeatedKFoldCrossValidation`, `PytorchLightningRepeatedKFoldCrossValidation`
- **feat** (1): `ChemPropFeaturizer`

## Proposed direction

Add a mixin carrying a `model_validator(mode="before")` that maps `random_state` onto `random_seed` and emits the `DeprecationWarning`, apply it to the classes above, and remove the `AnvilSection` hook. Both construction paths then behave identically, the two keys are collapsed before anything downstream can see both, and the `feature_base.py` docstring becomes accurate.

Constraints:

- **Defaults differ across these classes** (`split_base.py:59` and `tabpfn.py:54` default to `42`, most others to `None`), so the mixin should carry only the validator and leave each field declaration in place rather than declaring `random_seed` itself.
- **A pydantic `validation_alias`/`AliasChoices` alone would drop the warning.** It accepts the old name silently, so preserving the deprecation notice requires the before-validator. The repo uses no alias machinery today, so either approach introduces a new idiom.
- **Removing the spec hook changes behavior for classes lacking `random_seed`.** Those sections currently warn and discard; afterward `random_state` reaches the constructor, where `extra="ignore"` classes drop it silently and `extra="allow"` classes retain it as a stray attribute. Prefer rejecting it outright over either outcome.
- **`seed_to_sklearn_kwargs` should also stop keying off mere presence**, so that a `None` seed does not overwrite a value already present under the estimator's native name.

`architecture/model_base.py:17` (`seed_to_sklearn_kwargs`) is otherwise a separate and legitimate concern: mapping `random_seed` back to scikit-learn's native `random_state` when handing params to an estimator is correct and should stay.

## Relationship to #591

This is a concrete instance of both failure modes #591 describes, with a deprecated alias rather than a typo as the trigger. `ScaffoldSplitter(random_state=7)` is the silent drop (`extra="ignore"`); the model table above is the silent accept (`extra="allow"`), where the key survives validation and is then clobbered at build.

#591 proposes switching `rf.py`, `svm.py`, and `dummy.py` to `extra="allow"`. That change does not by itself fix the seed loss shown above, since the clobber happens in `seed_to_sklearn_kwargs` after validation. Sequencing this issue first collapses the two keys at the class boundary, so no model can carry both by the time `seed_to_sklearn_kwargs` runs.
